src/dipdup/indexes/tezos_tzkt_operations/matcher.py

```diff
@@ -194,9 +195,31 @@ def match_operation_subgroup(
                 pattern_index = 0
 
         if len(matched_operations) >= sum(0 if x.optional else 1 for x in handler_config.pattern):
-            _logger.info('%s: `%s` handler matched!', operation_subgroup.hash, handler_config.callback)
+            _logger.debug('%s: `%s` handler matched!', operation_subgroup.hash, handler_config.callback)
 
             args = prepare_operation_handler_args(package, handler_config, matched_operations)
             matched_handlers.append((operation_subgroup, handler_config, args))
 
-    return matched_handlers
+    if not (alt and len(matched_handlers) in (0, 1)):
+        return matched_handlers
+
+    # NOTE: Below is a secret algorithm for very special cases
```

Comment:
Add more context: this is about sorting matched handlers by the TzKT internal id (which is incremental) of the last operation in matched pattern